force-app/main/default/triggers/opportunityTrigger.trigger

```diff
@@ -0,0 +1,52 @@
+trigger opportunityTrigger on Opportunity (before update, before delete) {
+    // 1. Opportunity Update - Validate Amount > 5000
+    if (Trigger.isUpdate) {
+        for (Opportunity opp : Trigger.new) {
+            if (opp.Amount < 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+    
+    // 2. Prevent Deletion of Closed Won Opportunities for Banking Accounts
+    if (Trigger.isDelete) {
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : Trigger.old) {
+            if (opp.StageName == 'Closed Won') {
+                accountIds.add(opp.AccountId);
+            }
+        }
+
+        Map<Id, Account> accountMap = new Map<Id, Account>(
+            [SELECT Id, Industry FROM Account WHERE Id IN :accountIds]
+        );
+
+        for (Opportunity opp : Trigger.old) {
```

Comment:
What you did here works, but you can also query into a Map of Opportunities directly and use the relationship to Account like Account.Industry to check is banking. Then you can use this map in your compare to your original Trigger contact list.